src/emc/sai/saicanon.cc

```diff
@@ -519,147 +408,142 @@ void STRAIGHT_PROBE(int line_number,
          , b /*BB*/
          , c /*CC*/
          );
-  _probe_position_x = x;
-  _probe_position_y = y;
-  _probe_position_z = z;
-  _probe_position_a = a; /*AA*/
-  _probe_position_b = b; /*BB*/
-  _probe_position_c = c; /*CC*/
+  _sai._probe_position_x = x;
+  _sai._probe_position_y = y;
+  _sai._probe_position_z = z;
+  _sai._probe_position_a = a; /*AA*/
+  _sai._probe_position_b = b; /*BB*/
+  _sai._probe_position_c = c; /*CC*/
   if (distance != 0)
     {
-      backoff = ((_length_unit_type == CANON_UNITS_MM) ? 0.254 : 0.01);
-      _program_position_x = (x + (backoff * (dx / distance)));
-      _program_position_y = (y + (backoff * (dy / distance)));
-      _program_position_z = (z + (backoff * (dz / distance)));
+      backoff = ((_sai._length_unit_type == CANON_UNITS_MM) ? 0.254 : 0.01);
+      _sai._program_position_x = (x + (backoff * (dx / distance)));
+      _sai._program_position_y = (y + (backoff * (dy / distance)));
+      _sai._program_position_z = (z + (backoff * (dz / distance)));
     }
-  _program_position_a = a; /*AA*/
-  _program_position_b = b; /*BB*/
-  _program_position_c = c; /*CC*/
+  _sai._program_position_a = a; /*AA*/
+  _sai._program_position_b = b; /*BB*/
+  _sai._program_position_c = c; /*CC*/
 }
 
 
 void RIGID_TAP(int line_number, double x, double y, double z, double scale)
 {
-
-
-    fprintf(_outfile, "%5d ", _line_number++);
-    print_nc_line_number();
-    fprintf(_outfile, "RIGID_TAP(%.4f, %.4f, %.4f)\n", x, y, z);
-
+    ECHO_WITH_ARGS("%.4f, %.4f, %.4f", x, y, z);
 }
 
 
 void DWELL(double seconds)
-{PRINT1("DWELL(%.4f)\n", seconds);}
+{
+  ECHO_WITH_ARGS("%.4f", seconds);
+}
 
 /* Spindle Functions */
 void SPINDLE_RETRACT_TRAVERSE()
-{PRINT0("SPINDLE_RETRACT_TRAVERSE()\n");}
+{PRINT("SPINDLE_RETRACT_TRAVERSE()\n");}
 
 void SET_SPINDLE_MODE(int spindle, double arg) {
-  PRINT2("SET_SPINDLE_MODE(%d %.4f)\n", spindle, arg);
+  PRINT("SET_SPINDLE_MODE(%d %.4f)\n", spindle, arg);
 }
 
 void START_SPINDLE_CLOCKWISE(int spindle, int wait_for_atspeed)
 {
-  PRINT1("START_SPINDLE_CLOCKWISE(%i)\n", spindle);
-  _spindle_turning[spindle] = ((_spindle_speed == 0) ? CANON_STOPPED :
+  PRINT("START_SPINDLE_CLOCKWISE(%i)\n", spindle);
+  _sai._spindle_turning[spindle] = ((_spindle_speed == 0) ? CANON_STOPPED :
                                                    CANON_CLOCKWISE);
 }
 
 void START_SPINDLE_COUNTERCLOCKWISE(int spindle, int wait_for_atspeed)
 {
-  PRINT1("START_SPINDLE_COUNTERCLOCKWISE(%i)\n", spindle);
-  _spindle_turning[spindle] = ((_spindle_speed == 0) ? CANON_STOPPED :
+  PRINT("START_SPINDLE_COUNTERCLOCKWISE(%i)\n", spindle);
+  _spindle_turning = ((_spindle_speed == 0) ? CANON_STOPPED :
                                                    CANON_COUNTERCLOCKWISE);
 }
 
 void SET_SPINDLE_SPEED(int spindle, double rpm)
 {
-  PRINT2("SET_SPINDLE_SPEED(%i, %.4f)\n", spindle, rpm);
-  _spindle_speed[spindle] = rpm;
+  PRINT("SET_SPINDLE_SPEED(%i, %.4f)\n", spindle, rpm);
+  _spindle_speed = rpm;
 }
```

Comment:
Will these removal of [spindle] affect the multi-spindle stuff?  It looks at quick glance that before it was setting rpm and etc for the given spindle - so how are these changes handling it?